texttospeech.js

```diff
@@ -0,0 +1,25 @@
+
+function TTS() {
+	this.last_time = new Date().valueOf();
+	this.previous_text = null;
+}
+
+TTS.prototype = {
+	play: function(text) {
+		let now = new Date().valueOf();
+		let limit = this.last_time + 1000; 
+		if(text != this.previous_text || now > limit) {
+			console.log("tts.speak(" + text + ")");
+			window.speechSynthesis.cancel();
+			let u = new SpeechSynthesisUtterance();
+			u.text = text;
+			u.lang = 'ja-JP';
+			window.speechSynthesis.speak(u);
+			this.previous_text = text;
+			this.last_time = now;
+		} else {
+			console.log("Ignoring " + text);
```

Comment:
no need to do now, but maybe it'd be good to pop up a message if we know the reason why it didn't work (no tts library etc)

Reply:
(this was also mentioned before and I agree)